R/methods-filters.R

```diff
@@ -164,3 +164,16 @@ setMethod("filterEmptySpectra", "MSnExp",
               object <- logging(object, msg)
               object
           })
+
+setMethod("filterPrecursorScan", "MSnExp",
+          function(object, scanIndex, ...) {
+            object <- object[.filterSpectraHierarchy(fData(object), scanIndex)]
```

Comment:
This method will never work for `MSnExp` objects, because they don't have the required columns in the feature data. Any plans to implement it for `MSnExp` objects? A simple helper function that iterates through the `Spectrum` objects to extract the `acquisitionNum` and `precursorScanNum` and returns a `data.frame` might do it.

Reply:
Currently I don't have any plans to add such a function (because most `MSnExp` won't have more than one `msLevel`) but I used `MSnExp` to provide this option.

Reply:
Right, forgot about that.